repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -465,6 +498,13 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
       originalResponse.flushBuffer()
     }
 
+    if (sentError) {
```

Comment:
I think this needs to be higher, because as it stands you've potentially already committed the other body.